File: 2021/day1/part2.ts
#!/usr/bin/env npx ts-node

import * as fs from 'fs';
import { exit } from 'process';

const measurements = fs
    .readFileSync('./input.txt', 'utf-8')
    .split(/\n/)
    .filter(line => line != '')
    .map(numStr => parseInt(numStr));

const sum = (arr: number[]) => arr.reduce((pv, cv) => pv + cv, 0);

const windowSize = 3;
let increments = 0;
let prev = sum(measurements.slice(0, windowSize));

measurements.slice(windowSize).forEach((v, i) => {
    const current = prev + v - measurements[i];
    if (current > prev) increments += 1;
    prev = current;
});

console.log(increments);

File: 2017/day2/part2.py
#!/usr/bin/env python3


def get_quotient(values):
    for i in range(len(values)):
        for j in range(i+1, len(values)):
            if values[i] > values[j]:
                dividend, divisor = values[i], values[j]
            else:
                dividend, divisor = values[j], values[i]

            if dividend % divisor == 0:
                return dividend // divisor


quotient_sum = 0
with open('input.txt') as file:
    for line in file.readlines():
        values = list(map(int, line.strip().split("\t")))
        quotient_sum += get_quotient(values)

print(quotient_sum)
